webapp/src/test/java/com/box/l10n/mojito/android/strings/AndroidStringDocumentMapperTest.java

```diff
@@ -782,4 +783,461 @@ private TextUnitDTO textUnitDTO(
+    assertThat(plural.getComment()).isEqualTo("comment1");
+    assertThat(plural.getItems()).hasSize(5);
+    assertThat(plural.getItems().get(ZERO).getId()).isEqualTo(200L);
+    assertThat(plural.getItems().get(ZERO).getContent()).isEqualTo("content2_zero");
+    assertThat(plural.getItems().get(TWO).getId()).isEqualTo(202L);
+    assertThat(plural.getItems().get(TWO).getContent()).isEqualTo("content2_two");
+    assertThat(plural.getItems().get(FEW).getId()).isEqualTo(203L);
+    assertThat(plural.getItems().get(FEW).getContent()).isEqualTo("content2_few");
+    assertThat(plural.getItems().get(MANY).getId()).isEqualTo(204L);
+    assertThat(plural.getItems().get(MANY).getContent()).isEqualTo("content2_many");
+    assertThat(plural.getItems().get(OTHER).getId()).isEqualTo(205L);
+    assertThat(plural.getItems().get(OTHER).getContent()).isEqualTo("content2_other");
+  }
+
+  @Test
+  public void testReadFromSourceTextUnitsWithDuplicatedOnePlurals() {
+    mapper = new AndroidStringDocumentMapper(" _", assetDelimiter);
+    textUnits.add(sourceTextUnitDTO(123L, "name0", "content0", "comment0", "my/path0", null, null));
+
+    textUnits.add(
+        sourceTextUnitDTO(
+            100L,
+            "name1 _other",
+            "content1_zero",
+            "comment1",
+            "my/path0.xml",
+            "zero",
+            "name1_other"));
+    textUnits.add(
+        sourceTextUnitDTO(
+            101L,
+            "name1 _other",
+            "content1_one",
+            "comment1",
+            "my/path0.xml",
+            "one",
+            "name1_other"));
+    textUnits.add(
+        sourceTextUnitDTO(
+            102L,
+            "name1 _other",
+            "content1_two",
+            "comment1",
+            "my/path0.xml",
+            "two",
+            "name1_other"));
+    textUnits.add(
+        sourceTextUnitDTO(
+            103L,
+            "name1 _other",
+            "content1_few",
+            "comment1",
+            "my/path0.xml",
+            "few",
+            "name1_other"));
+    textUnits.add(
+        sourceTextUnitDTO(
+            104L,
+            "name1 _other",
+            "content1_many",
+            "comment1",
+            "my/path0.xml",
+            "many",
+            "name1_other"));
+    textUnits.add(
+        sourceTextUnitDTO(
+            105L,
+            "name1 _other",
+            "content1_other",
+            "comment1",
+            "my/path0.xml",
+            "other",
+            "name1_other"));
+
+    textUnits.add(
+        sourceTextUnitDTO(
+            201L,
+            "name1 _other",
+            "content1_one",
+            "comment1",
+            "my/path0.xml",
+            "one",
+            "name1_other"));
+
+    assertThrows(
+        AndroidPluralDuplicateKeyException.class, () -> mapper.readFromSourceTextUnits(textUnits));
+  }
+
+  @Test
+  public void testReadFromSourceTextUnitsWithDuplicatedOtherPlurals() {
```

Comment:
This is the test for when the other plural is duplicated (including the same source)